WalletWasabi/Models/TransactionBuilding/FeeStrategy.cs

```diff
@@ -0,0 +1,79 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Models.TransactionBuilding
+{
+	public class FeeStrategy
+	{
+		private int _target;
+		private FeeRate _rate;
+
+		public FeeStrategyType Type { get; }
+
+		public int Target
+		{
+			get
+			{
+				if (Type != FeeStrategyType.Target)
+				{
+					throw new NotSupportedException($"Cannot get {nameof(Target)} with {nameof(FeeStrategyType)} {Type}.");
+				}
+				return _target;
+			}
+		}
+
+		public FeeRate Rate
+		{
+			get
+			{
+				if (Type != FeeStrategyType.Rate)
+				{
+					throw new NotSupportedException($"Cannot get {nameof(Target)} with {nameof(FeeStrategyType)} {Type}.");
```

Comment:
Maybe the message should say "Cannot get {nameof(`Rate`)} with ....

Reply:
Yes, definitely! Fixed.